File: selection_sort/sort.py
from timeit import default_timer as timer

def get_words():
  fname = "../dataset/words_10000.txt"

  wordlist = []

  with open(fname, "r") as fd:
    for line in fd:
      wordlist.append(line.strip())

  return wordlist

# O(n^2)
def selection_sort(wordlist):
  length = len(wordlist)

  for i in range(length):
    for j in range(i+1, length):
      if wordlist[i] > wordlist[j]:
        aux = wordlist[i]
        wordlist[i] = wordlist[j]
        wordlist[j] = aux

if __name__ == "__main__":
  wordlist = get_words()
  wordlist2 = get_words()

  print(wordlist[:10], wordlist[-10:])

  tsort = timer()
  selection_sort(wordlist)
  tsort = timer() - tsort

  tsort2 = timer()
  wordlist2.sort()
  tsort2 = timer() - tsort2

  print(wordlist[:10], wordlist[-10:])
  print(wordlist2[:10], wordlist2[-10:])
  print(tsort, tsort2)

File: selection_sort/sort.js
const fs = require('fs')

function get_words() {
  const fname = "../dataset/words_100000.txt";
  
  return fs.readFileSync(fname, 'utf8').split('\n').filter(Boolean);
}

function selection_sort(list) {
  for (let i = 0; i < list.length; i++) {
    for (let j = i + 1; j < list.length; j++) {
      if (list[i] > list[j]) {
        let aux = list[i];
        list[i] = list[j];
        list[j] = aux;
      }
    }
  }
}

let wordlist = get_words();
let wordlist2 = get_words();

console.log(wordlist);

var tsort = process.hrtime.bigint();
selection_sort(wordlist);
tsort = (process.hrtime.bigint() - tsort);

var tsort2 = process.hrtime.bigint();
wordlist2.sort();
tsort2 = (process.hrtime.bigint() - tsort2);

console.log(wordlist);
console.log(wordlist2);
console.log(tsort, tsort2);
